### ISSUE TITLE
Add ruby environment instructions to Install guide

### ISSUE THREAD

**Issue by tedsanders** [jekyll/jekyll#6800]
- [ ] I believe this to be a bug, not a question about using Jekyll.
- [ ] I updated to the latest Jekyll (or) if on GitHub Pages to the latest `github-pages`
- [ ] I ran `jekyll doctor` to check my configuration
- [ ] I read the CONTRIBUTION file at https://jekyllrb.com/docs/contributing/
- [ ] This is a feature request.

---

- [ ] I am on (or have tested on) ***macOS*** 10+
- [X] I am on (or have tested on) ***Debian/Ubuntu*** GNU/Linux
- [ ] I am on (or have tested on) ***Fedora*** GNU/Linux
- [ ] I am on (or have tested on) ***Arch*** GNU/Linux
- [ ] I am on (or have tested on) ***Other*** GNU/Linux
- [ ] I am on (or have tested on) ***Windows*** 10+

---

- [X] I was trying to install.
- [ ] There is a broken Plugin API.
- [ ] I had an error on GitHub Pages, and I have reproduced it locally.
- [ ] I had an error on GitHub Pages, and GitHub Support said it was a Jekyll Bug.
- [ ] I had an error on GitHub Pages and I did not test it locally.
- [ ] I was trying to build.
- [ ] It was another bug.

## My Reproduction Steps

This isn't a Jekyll bug, but a suggestion for improving the install experience for new users. Hopefully the suggestion won't take much time or thought to implement.

I have a fresh install of Linux Mint 17. Coming from using Jekyll in a Windows environment.

Went first to the install page: https://jekyllrb.com/docs/installation/#install-with-rubygems

Checked the everything was installed.

Ran the very first command to install Jekyll (gem install Jekyll).

Immediately was told by terminal that I don't have write permissions. Googled around and saw that this is a common problem among many users and distros.

I'm not saying this is a bug with Jekyll, it's more about setting up Ruby environments, but I just wanted to highlight that it's a giant user barrier when Linux users run into an error on the very first step of an installation guide. Many users thinking of trying Jekyll out may just stop here.

Therefore, I suggest adding a link or instruction or paragraph that explains the best way to get around the permissions issue. I think this will strengthen the pipeline of new Jekyll users and reduced headaches for everyone. Thanks for your time and sorry if I've said anything ignorant.

**Comment by tedsanders** [comment]
After more than an hour, I still don't have Jekyll installed because now I'm trying to read how RVM works to see if installing a second copy of Ruby will somehow get around the permissions issue.

**Comment by tedsanders** [comment]
I did search around and find this Jekyll talk page: https://talk.jekyllrb.com/t/you-dont-have-write-permissions-for-the-var-lib-gems-2-3-0-directory/204/4

The answer there seemed very unsatisfying and unhelpful to a new user. Saying that it's not a problem with Jekyll and therefore we shouldn't help users during installation doesn't seem user-friendly to me.

**Comment by tedsanders** [comment]
(Ironically I tried following the instructions in that talk thread, which said to install Bundle. But then I ran into the exact same permissions issue! :D )

**Comment by tedsanders** [comment]
Update: I wanted to try drafting a suggestion of steps to include, but I've spent hours on this and still haven't gotten past step one of installation. I tried installing both Bundle and RVM, but neither got me around the permissions issue. I never realized that Jekyll on Windows would be 10x easier than Jekyll on Linux. :)

**Comment by Crunch09** [comment]
Hi @tedsanders 👋 ,
i think it's hard for jekyll to include specific ruby installation instructions as there are many different ways to install ruby and we would basically duplicate installation instructions from dedicated ruby installers. That said, we're always looking for ways to make it easier to use Jekyll so thank you for your feedback!

> I tried installing both Bundle and RVM,
these are not the same:
- `bundle(r)` is a ruby gem which manages gem dependencies
- `rvm` is a ruby version manager

I assume you have tried the installation instruction for RVM on [http://rvm.io/](http://rvm.io/) ?

**Comment by mkasberg** [comment]
I can see how it would be really frustrating for a new user to follow the instructions exactly as they are written and get errors. I think it would be better if the instructions worked as they are written, or at least if there was an info bubble explaining some of the alternatives... Maybe I could get a pull request together.

@tedsanders Did you get it working yet? Personally, I think the best way to get started quickly is like this with bundler, which keeps all dependencies inside the project folder:

```
$ cd my-jekyll-project
$ bundle init                          # Creates an empty Gemfile
$ bundle install --path vendor/bundle  # Set install path to vendor/bundle sub-folder
$ bundle add jekyll                    # Add & Install Jekyll Gem
```

Then, run your Jekyll commands inside `bundle exec`, like `bundle exec jekyll serve`.

**Comment by tedsanders** [comment]
@Crunch09 , yes Bundler and RVM are not the same. But they were the two solutions given on the Jekyll talk page I linked to, which I why I tried both. And yes, I did follow the instructions on rvm.io when installing RVM.

My eventual solution was to use a login-shell (as opposed to a non-login shell?). Using a login shell to install RVM and Bundler and jekyll removed the permissions problem. Perhaps that could be added to the instructions, if it sounds like a general problem.

Even if it costs 100 people 5 seconds, it's worth it if it saves one person a few hours.

**Comment by DirtyF** [comment]
I don't run GNU/Linux, I wish there was an easy 3 steps install to install Jekyll on a majority of Linux distros. Will happily review a PR if you think that's more easy for a majority of users.

**Comment by mkasberg** [comment]
I created a pull request to update the docs, let me know what you think.

I proved to myself that this works on a fresh Ubuntu installation by testing with the following Dockerfile:

```docker
FROM ubuntu:16.04

RUN apt-get update
RUN apt-get install -y ruby ruby-dev build-essential && \
    gem install bundler
RUN useradd --create-home jekyll

USER jekyll
RUN mkdir -p /home/jekyll/project
WORKDIR /home/jekyll/project
RUN bundle init && bundle install --path vendor/bundle && bundle add jekyll

CMD ["bundle", "exec", "jekyll", "serve"]
```

I believe all of the Bundler instructions will work equally well on macOS, though I don't have a Mac to test it on.

**Comment by mkasberg** [comment]
There was some really good discussion about all of this in #6828, but we ultimately decided that a tutorial was the best location for the Bundler instructions (thanks for the merge!). I think that will be a great tutorial, but I also think it is worth considering some additional improvements to the Installation page, since it's where most users land and still seems to be confusing - #6851, #6852.

Is anyone working on this? @DirtyF had some good proposals in the discussion of #6828, particularly around different sections for Mac/Linux instructions... I'd be happy to contribute a little more if we have an idea which direction to head.

**Comment by DirtyF** [comment]
@mkasberg I have only something ready for macOS, feel free to work on the Linux part.

**Comment by PJOssenbruggen** [comment]
I have a goal to establish a GitHub Pages website using jekyll theme: "Tactile" for MacOS. As you can see, I am unable to get past step 1 of https://jekyllrb.com:

```
PJO:~ PJO$ gem install jekyll bundler
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.3.0 directory.
```

I have tried several approaches to get past this step over the past two days without success. https://github.com/jekyll/jekyll/issues/6892. If you read my posting, you can tell that I am a novice. 

Anyway, I thank @mkasberg for offering to contribute his time tackling this issue. 

In the meantime, I will look at @DirtyF's #6823 and other postings.

I thank all of you.
Paul

**Comment by DirtyF** [comment]
@tedsanders Did you manage to solve the permissions issue. How? 

What should we include for Linux Users so that they can follow instructions that actually work ?

**Comment by mkasberg** [comment]
I think the best approach for Linux (barring Bundler) is probably to do a "user install" of the gem. That can be done with `gem install jekyll --user-install`. Or, setting `GEM_HOME` to something like `~/.gem` will have a similar effect. With either of these approaches, it's also necessary to make sure the user's `PATH` is set up correctly.

[This](https://github.com/rubygems/rubygems/issues/1394) discussion provides a little more insight into the issue.

There are other approaches to consider as well, but I don't think they're as good and probably shouldn't be recommended as the best way to go. For example, Ubuntu users can simply `sudo apt install jekyll`. But then they are installing a gem with apt-get (kind of weird), and the version that Ubuntu ships usually lacks behind the latest stable version.

**Comment by DirtyF** [comment]
@mkasberg https://jekyllrb.com/docs/installation/#macOS :)

**Comment by tedsanders** [comment]
@DirtyF: I ended up getting around the permissions issue (I think) by using a login shell instead of the standard shell in Linux Mint.

Also happy to report that I tried installing Jekyll today on a fresh Ubuntu install and ran into zero problems. Thanks everyone for your hard work on Jekyll. Such a great project for the world.
